src/perl/lib/wtsi_clarity/util/validators.pm

```diff
@@ -0,0 +1,136 @@
+package wtsi_clarity::util::validators;
+
+use strict;
+use warnings;
+
+use wtsi_clarity::util::validation;
+
+use List::Util   qw/reduce/;
+use Scalar::Util qw/looks_like_number/;
+use Readonly;
+use Exporter 'import';
+
+our @EXPORT_OK = qw/flgen_bc ean13_bc/;
+
+our $VERSION = '0.0';
+
+Readonly::Scalar my $FLUIDIGM_BC_LENGTH => 10;
+Readonly::Scalar my $EAN_BC_LENGTH      => 13;
+
+sub flgen_bc {
+  return _validator(_has_length($FLUIDIGM_BC_LENGTH), _is_integer())->(shift);
+}
+sub ean13_bc {
+  return _validator(_has_length($EAN_BC_LENGTH), _is_integer())->(shift);
+}
+
+sub _validator {
+  my @validators = @_;
+
+  return sub {
+    my $val = shift;
+
+    my $errors = reduce {
+      if (!$b->{'check'}->($val)) {
+        push @{$a}, $b->{'message'};
+      }
+      return $a;
+    } [], @validators;
+
+    return wtsi_clarity::util::validation->new(value => $val, errors => $errors);
+  }
+}
+
+sub _has_length {
+  my $length = shift;
+  my $message = shift;
+
+  return {
+    check   => sub {
+      my $val = shift;
+      return $length == length $val;
+    },
+    message => $message // 'The barcode must have a length of ' . $length,
+  }
+}
+
+sub _is_integer {
+  my $message = shift;
+
+  return {
+    check   => \&looks_like_number,
+    message => $message // 'The barcode must be numeric',
+  }
+}
+
+1;
+
+__END__
+
+=head1 NAME
+
+wtsi_clarity::util::validators
+
+=head1 SYNOPSIS
+
+use wtsi_clarity::util::validators qw/flgen_bc/;
+
+my $validation = flgen_bc('1234567');
+
+if (!$validation->failed) {
+  croak $validation->error_message;
+}
+
+=head1 DESCRIPTION
+
+A module for providing a couple of different validations, each of which return a Validation
+object with 2 methods on: failed and error_message.
+
```

Comment:
I guess this solution is working, but this module is doing 'too much'.
It contains validators and it is also doing validations. I think, it is breaking the single responsibility principle.
It would be better to separate them.